frontend/src/components/AddClassModal.tsx

```diff
@@ -220,20 +199,49 @@ export class AddClassModal extends React.Component<
         {this.state.queuedCourses.map(
           (course: ScheduleCourse, index: number) => {
             return (
-              <QueuedClass>
-                <p style={{ width: 90 }}>{course.subject + course.classId}</p>
-                <p>{course.name}</p>
-                <XButton
-                  onClick={this.removeQueuedClass.bind(this, index)}
-                ></XButton>
-              </QueuedClass>
+              <NonDraggableClassBlock
+                course={course}
+                onDelete={this.removeQueuedClass.bind(this, index)}
+              />
             );
           }
         )}
       </QueuedClassesWrapper>
     );
   }
 
+  renderSubjectTextField() {
+    return (
+      <TextField
+        id="outlined-basic"
+        label="Subject"
+        variant="outlined"
+        value={this.state.formSubject}
+        onChange={this.handleSubjectChange.bind(this)}
+        placeholder="CS"
+        error={this.state.formSubject.length === 0}
+        style={{ marginTop: 36, minWidth: 326 }}
+        color="secondary"
+      />
+    );
+  }
+
+  renderClassIDTextField() {
+    return (
+      <TextField
+        id="outlined-basic"
+        label="Class ID"
+        variant="outlined"
+        value={this.state.formClassId}
+        onChange={this.handleClassIdChange.bind(this)}
+        placeholder="3500"
+        error={this.state.formClassId.length === 0}
```

Comment:
Right now it starts off in the error state. I'd say it doesn't really need to error at all, because we add error text when the request isn't filled out properly (same with the subject text field)